### PR TITLE
support dynamic shape export to onnx

### PR DESCRIPTION
**Pull request by QingChuanWS**
This pr focus on support onnx dynamic shape export.

### CODE REVIEW

File: mmcls/models/classifiers/image.py [hunk 1]
```diff
@@ -77,7 +77,7 @@ def forward_train(self, img, gt_label, **kwargs):
 
         return losses
 
-    def simple_test(self, img, img_metas):
+    def simple_test(self, img):
```
Comment (RunningLeon) [inline mmcls/models/classifiers/image.py:80]: Why remove `img_metas`?  better not change the original interface.

File: tools/pytorch2onnx.py [hunk 2]
```diff
@@ -65,14 +66,31 @@ def pytorch2onnx(model,
     # replace original forward function
     origin_forward = model.forward
     model.forward = partial(model.forward, return_loss=False)
-
     register_extra_symbolics(opset_version)
+
+    # support dynamic shape export
+    if dynamic_shape:
+        dynamic_axes = {
```
Comment (RunningLeon) [inline tools/pytorch2onnx.py:73]: Add `batch` dynamic part as `0:'batch'` to inputs and outputs

File: tools/pytorch2onnx.py [hunk 3]
```diff
@@ -65,14 +66,31 @@ def pytorch2onnx(model,
...
+            'input': {
+                2: 'width',
+                3: 'height'
+            },
+            'labels': {
+                0: 'num_img'
```
Comment (RunningLeon) [inline tools/pytorch2onnx.py:79]: Change `num_img` to `batch`

File: tools/pytorch2onnx.py [hunk 4]
```diff
@@ -65,14 +66,31 @@ def pytorch2onnx(model,
...
+        }
+    else:
+        dynamic_axes = {}
+
     with torch.no_grad():
         torch.onnx.export(
             model, (img_list, ),
             output_file,
+            input_names=['input'],
+            output_names=['labels'],
```
Comment (RunningLeon) [inline tools/pytorch2onnx.py:90]: output name should be `probs`.

File: tools/pytorch2onnx.py [hunk 5]
```diff
@@ -119,6 +144,10 @@ def parse_args():
         nargs='+',
         default=[224, 224],
         help='input image size')
+    parser.add_argument(
+        '--dynamic-shape',
+        action='store_true',
+        help='export dynamic onnx graph')
```
Comment (RunningLeon) [inline tools/pytorch2onnx.py:150]: Change to `Whether to export ONNX with dynamic input shape. Defaults to False.`

File: mmcls/models/classifiers/image.py (hunk 1 above)
Comment (QingChuanWS) [reply mmcls/models/classifiers/image.py:80]: If i do not remove this option, pytorch2onnx.py will report an error when running.

File: tools/pytorch2onnx.py (hunk 5 above)
Comment (QingChuanWS) [reply tools/pytorch2onnx.py:150]: right!

File: mmcls/models/classifiers/image.py (hunk 1 above)
Comment (QingChuanWS) [reply mmcls/models/classifiers/image.py:80]: I try to modifying pytorch2onnx.py to avoiding change the original code.

**Comment by RunningLeon** [comment]
LGTM.

**Review by ycxioooong — APPROVED** [review]
LGTM

**Pull request merged**